server/database/models/TransactionRepository.js

```diff
@@ -56,6 +56,61 @@ class TransactionRepository extends AbstractRepository {
     return rows[0];
   }
 
+  async expensesAmount(userId, periodFilter) {
+    let queryFilter = "";
+    if (periodFilter === "day") {
+      queryFilter = `AND DATE(t.date) = CURDATE()`;
+    }
+    if (periodFilter === "week") {
+      queryFilter = `AND WEEK(t.date, 1) = WEEK(CURDATE(), 1)
+      AND YEAR(t.date) = YEAR(CURDATE())`;
+    }
+    if (periodFilter === "month") {
+      queryFilter = `AND MONTH(t.date) = MONTH(CURDATE())
+      AND YEAR(t.date) = YEAR(CURDATE())`;
+    }
+    if (periodFilter === "year") {
+      queryFilter = `AND YEAR(t.date) = YEAR(CURDATE())`;
+    }
+
+    const [rows] = await this.database.query(
+      `SELECT SUM(amount) expenses_amount
+       FROM ${this.table} t
+       WHERE t.user_id = ? ${queryFilter}`,
+      [userId]
+    );
+    return rows[0].expenses_amount;
+  }
```

Comment:
**question**: Explain me how it works, give me multiple examples of inputs and expected outputs

Reply:
Possibles inputs are day, week, month or year.

If the input is day, it will return the sum of all actual day's transactions. And only the amount.

If the input is week, it will return the sum of all actual week's transactions.
Here the `AND YEAR(t.date) = YEAR(CURDATE())` is used to target actual year.
The MySQL week function starts naturally weeks at sunday, so i add to put a parameter in `WEEK(t.date, 1)` , which is 1, for it to start weeks at monday.

If the input is month and year, it works mostly the same way, returning either the sum of all actual month's transactions or year's transactions.

Reply:
Okay, then it only works for current day/week/month/year ? An user cannot request transactions from january for example ?

It seems odd, I would expect the ability to select a period for my dashboard in order to understand my budget.

Reply:
Yes i know, it will added later since i will not have time to implement it right now.